packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx

```diff
@@ -130,10 +130,11 @@ export const PatientRegistration: React.FC<PatientRegistrationProps> = ({ savePa
         kind: 'success',
       });
 
-      const redirectUrl = interpolateUrl(
-        new URLSearchParams(search).get('afterUrl') ||
-          interpolateString(config.links.submitButton, { patientUuid: values.patientUuid }),
-      );
+      const afterUrl = new URLSearchParams(search).get('afterUrl');
+      const includePatientUuidInAfterUrl = new URLSearchParams(search).get('includePatientUuidInAfterUrl');
+      const redirectUrl = afterUrl
+        ? `${afterUrl}${includePatientUuidInAfterUrl === 'true' && `?patientUuid=${values.patientUuid}`}`
```

Comment:
It might be best to use [`interpolateUrl`](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-framework/docs/API.md#interpolateurl) here. That way, instead of needing this much logic, you could just call:

`interpolateUrl(afterUrl, { patientUuid: values.patientUuid })`

And if the URL has `${patientUuid}` in it, it will automatically be added...

Reply:
Thanks guys. TBH I wasn't sure about how `interpolateUrl` and `interpolateString` were supposed to be used in conjunction with the `navigate({to: ...`. It's a pretty funky syntax and I didn't have the time to guess at it, especially this code is putting a redirect inside of another redirect. Let me figure it out and update the API.md

Reply:
Really, the difference between `interpolateUrl()` and `interpolateString()` is just that `interpolateUrl()` predefines `openmrsBase` and `openmrsSpaBase`... and hey! What are reviews for if not being pedantic about things you didn't have time to address 😁.

Reply:
No, I appreciate Eagle Eye as always. Nice catch @manuelroemer and @ibacher